docs/spec/efficient_querying.rst

```diff
@@ -0,0 +1,96 @@
+.. _efficient_querying:
+
+==================
+Efficient Querying
+==================
+
+In general Kartothek should know as early as possible which data to exclude, so
+Kartothek doesn't have to read and transfer it. If you're going to query a
+dataset a lot you should consider storing it in an optimized layout first. See
+:ref:`writing`.
+
+
+Background
+----------
+
+To understand how to query efficiently you should understand Kartothek's
+querying process:
+
+When querying data, Kartothek tries to apply the given predicates as early as
+possible, so as few parquet files as possible are read and parsed:
+
+- Kartothek looks if there exists an index for the filtered column. This is
+  either if the column is partitioned (primary index) or a secondary index was
+  generated. The results are combined for multiple columns.
+- In the resulting set of Parquet files, Kartothek looks at the footer, which
+  states the range of values for each column in this file. If there are multiple
+  row groups each one specifies the range of values inside it. Kartothek checks
+  if the queried data is in the range (of a row group) and therefore if this
+  row group is loaded. This process is called Predicate Pushdown.
+
+Kartothek then downloads these row groups and filters
+the resulting Pandas dataframes for the conditions that could not be
+(completely) resolved by then.
+
+More information on Parquet can be found in the
+`Parquet Documentation <https://parquet.apache.org/documentation/latest/>`_.
+
+
+Reading
+-------
+
+Following the dataset's :ref:`filesystem_layout` and querying process you should
+include the following in your query:
+
+* restrictions on possible partitions
+* restrictions on indexed columns
```

Comment:
I think this is more correct
`* restrictions on indexed columns (columns on which the dataset is partitioned or secondary indices)`